### PR TITLE
fix(9966): fix creation of multiple KVStore watchers for CNPs and CCNPs

### LINKED ISSUE #9966: cilium-operator creates 2 kvstore watchers per CNP or CCNP created
1st watcher which is used to have a shared store

https://github.com/cilium/cilium/blob/7393c2c302f428b37613724e6c0539c54f0a0f84/operator/cnp_event.go#L66-L71

2nd watcher which is used to handle events from the kvstore

https://github.com/cilium/cilium/blob/25ef8ab702409b783ded40d419875eea52b6584f/pkg/k8s/cnpstatus.go#L122-L127

Perhaps we should have something similar to the k8s api where we have a store and handlers that can be called whenever that store has an element added / updated / deleted

### PR DESCRIPTION
**Pull request by fristonio**
* Fixes #9966
* Implement Observer for CNPStatusEventHandler to remove the need of extra
  KVstore watcher.

### CODE REVIEW

**Comment by fristonio** [comment]
test-me-please

File: operator/ccnp_event.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ func enableCCNPWatcher() error {
 	}
 
 	if kvstoreEnabled() {
+		ccnpStatusMgr = k8s.NewCCNPStatusEventHandler(nil, ccnpStore, operatorOption.Config.CNPStatusUpdateInterval)
 		ccnpSharedStore, err := store.JoinSharedStore(store.Configuration{
```
Comment (aanm) [inline operator/ccnp_event.go:61]: Why doesn't this `ccnpSharedStore` have  observer?

File: pkg/k8s/ccnpstatus.go [hunk 2]
```diff
@@ -44,14 +44,3 @@ func NewCCNPStatusEventHandler(cnpStore *store.SharedStore, k8sStore cache.Store
 		CNPStatusEventHandler: NewCNPStatusEventHandler(cnpStore, k8sStore, updateInterval),
 	}
 }
-
```
Comment (aanm) [inline pkg/k8s/ccnpstatus.go:47]: On line 42, `func NewCCNPStatusEventHandler` no longer uses `cnpStore` so we can remove it from the function arguments.

File: operator/cnp_event.go [hunk 3]
```diff
@@ -68,19 +68,19 @@ func enableCNPWatcher(apiextensionsK8sClient apiextensionsclientset.Interface) e
...
+			Observer: cnpStatusMgr,
 		})
 		if err != nil {
 			return err
 		}
 
-		cnpStatusMgr = k8s.NewCNPStatusEventHandler(cnpSharedStore, cnpStore, cnpStatusUpdateInterval)
-
-		go cnpStatusMgr.WatchForCNPStatusEvents()
+		cnpStatusMgr.UpdateCNPStore(cnpSharedStore)
```
Comment (aanm) [inline operator/cnp_event.go:86]: Please add a comment that is safe to update the store here given that the store will only be used by the `StartStatusHandler` function. (Same for the CCNP similar code)

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by fristonio** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**